src/main/java/uk/gov/hmcts/darts/common/service/impl/HearingCommonServiceImpl.java

```diff
@@ -69,4 +71,47 @@ private HearingEntity setHearingLastDateModifiedBy(final HearingEntity hearingEn
         hearingEntity.setLastModifiedBy(userAccountEntity);
         return hearingRepository.saveAndFlush(hearingEntity);
     }
+
+    /**
+     * Links an audio media entity to a hearing if it exists.
+     *
+     * @param courtCaseEntity the court case entity associated with the hearing
+     * @param mediaEntity     the media entity to link
+     * @return true if the media was successfully linked, false otherwise
+     */
+    @Override
+    public boolean linkAudioToHearings(
+        CourtCaseEntity courtCaseEntity,
+        MediaEntity mediaEntity) {
+        if (courtCaseEntity == null) {
+            log.info("Can not link hearing to media {} as CourtCaseEntity is null", mediaEntity.getId());
+            return false;
+        }
+
+        Optional<HearingEntity> hearingEntityOptional = hearingRepository
+            .findHearing(
+                courtCaseEntity,
+                mediaEntity.getCourtroom(),
+                mediaEntity.getStart().toLocalDate()
```

Comment:
Will this link the audio to _any_ hearing that occurred on that day? Not sure that is what we want?

Reply:
Maybe this is fine because you're searching by court case.